auth/range_perm_cache.go

```diff
@@ -146,18 +87,44 @@ func checkKeyPerm(cachedPerms *unifiedRangePermissions, key, rangeEnd []byte, pe
 		plog.Panicf("unknown auth type: %v", permtyp)
 	}
 
-	requiredPerm := &rangePerm{begin: key, end: rangeEnd}
+	ivl := adt.NewStringAffineInterval(key, rangeEnd)
 
-	for _, perm := range tocheck {
-		if isSubset(requiredPerm, perm) {
-			return true
+	minBegin := adt.StringComparable("")
+	maxEnd := adt.StringComparable("\xff")
+
+	tocheck.Visit(ivl, func(n *adt.IntervalValue) bool {
+		if n.Ivl.Begin.Compare(minBegin) < 0 {
+			minBegin = n.Ivl.Begin.(adt.StringComparable)
+		}
+
+		if maxEnd.Compare(n.Ivl.End) < 0 {
+			maxEnd = n.Ivl.End.(adt.StringComparable)
 		}
+
```

Comment:
The word "continuous" wasn't good. I just tried to mean "no hole in the granted permissions". The check would be performed like this:
```
	tocheck.Visit(ivl, func(n *adt.IntervalValue) bool {
		visited = append(visited, n)

		return false
	})
	for i := 0; i < len(visited)-1; i++ {
		if visited[i].Ivl.End.Compare(visited[i+1].Ivl.Begin) == -1 {
			// there is a hole, permission denied
			return false
		}
	}
```

The loop can handle the case you showed?